src/Concretize.hs

```diff
@@ -677,26 +678,29 @@ concretizeDefinition allowAmbiguity typeEnv globalEnv visitedDefinitions definit
 -- For all other functions, the name must match exactly, and in all cases, the signature must match.
 allImplementations :: TypeEnv -> Env -> String -> Ty -> [(Env, Binder)]
 allImplementations typeEnv env functionName functionType =
-  filter (predicate . xobjTy . binderXObj . snd) foundBindings
+  (filter (predicate . xobjTy . binderXObj . snd) foundBindings)
   where
     predicate (Just t) =
       --trace ("areUnifiable? " ++ show functionType ++ " == " ++ show t ++ " " ++ show (areUnifiable functionType t)) $
       areUnifiable functionType t
     predicate Nothing = error "allfunctionswithnameandsignature"
-    foundBindings = case lookupBinder (SymPath [] functionName) (getTypeEnv typeEnv) of
+    foundBindings = case getTypeBinder typeEnv functionName of
       -- this function is an interface; lookup implementations
-      Just (Binder _ (XObj (Lst (XObj (Interface _ paths) _ _ : _)) _ _)) ->
+      Right (Binder _ (XObj (Lst (XObj (Interface _ paths) _ _ : _)) _ _)) ->
         -- N.B./TODO: There are functions designed for this
```

Comment:
This comment maybe can be changed now? :)

Reply:
Yes, sg!